Installation issue: dorado

### Steps to reproduce the issue

```yaml
spack:
  specs:
    - dorado@0.5.3+cuda
  concretizer:
    unify: true
    targets:
    # Determine whether we want to target specific or generic
    # microarchitectures. Valid values are: "microarchitectures" or "generic".
    # An example of "microarchitectures" would be "skylake" or "bulldozer",
    # while an example of "generic" would be "aarch64" or "x86_64_v4".
      granularity: generic
    # If "false" allow targets that are incompatible with the current host (for
    # instance concretize with target "icelake" while running on "haswell").
    # If "true" only allow targets that are compatible with the host.
      host_compatible: true
  container:
    format: docker
    images:
      os: "ubuntu:20.04"
      spack: "latest"
```


### Error message

<details><summary>Error message</summary>

<pre>
2376.1 1 error found in build log:
2376.1      254      CMakeFilters.cmake:176 (find_package)
2376.1      255      CMakeLists.txt:923 (include)
2376.1      256    This warning is for project developers.  Use -Wno-dev to suppress i
2376.1             t.
2376.1      257
2376.1      258    -- Could NOT find libaec: found neither libaecConfig.cmake nor liba
2376.1             ec-config.cmake (found version "1.0.6")
2376.1      259    -- Filter PLUGIN file is /
2376.1   >> 260    CMake Error at CMakeLists.txt:1183 (message):
2376.1      261       **** Parallel and C++ options are mutually exclusive, override w
2376.1             ith HDF5_ALLOW_UNSUPPORTED option ****
2376.1      262
2376.1      263
2376.1      264    -- Configuring incomplete, errors occurred!
</pre></details>


### Information on your system

The box is stopped. Will add later.

### Additional information

c7.4xl instance on AWS. I tried to build it with `ubuntu` and `rockylinux:9` and natively on the instance running Amazon Linux 2. Always with the same outcome. :( 

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/dorado/package.py
+++ b/var/spack/repos/builtin/packages/dorado/package.py
@@ -26,7 +26,7 @@ class Dorado(CMakePackage, CudaPackage):
     depends_on("git", type="build")
     depends_on("curl", type="build")
     depends_on("cuda")
-    depends_on("hdf5@1.17:+hl+cxx+szip")
+    depends_on("hdf5@:1+hl+cxx+szip")
     depends_on("htslib@1.15.1")
     depends_on("openssl")
     depends_on("zstd")

